plugins/edsm.py

```diff
@@ -331,87 +471,116 @@ def cmdr_data(data, is_beta):
 
 
 # Worker thread
-def worker():
+def worker() -> None:
+    """
+    Upload worker.
 
-    pending = []	# Unsent events
+    Works based the `this.queue` queue, loops infinitely until the item returned from the queue is None.
+    """
+    pending = []  # Unsent events
     closing = False
 
     while True:
-        item = this.queue.get()
+        item: Optional[Tuple[str, Mapping[str, Any]]] = this.queue.get()
         if item:
             (cmdr, entry) = item
         else:
-            closing = True	# Try to send any unsent events before we close
+            closing = True  # Try to send any unsent events before we close
 
         retrying = 0
         while retrying < 3:
             try:
-                if item and entry['event'] not in this.discardedEvents:
+                if TYPE_CHECKING:
+                    # Tell the type checker that these two are bound.
+                    # TODO: While this works because of the item check below, these names are still technically unbound
+                    # TODO: in some cases, therefore this should be refactored.
+                    cmdr: str = ""
+                    entry: Mapping[str, Any] = {}
+
+                if item and entry['event'] not in this.discardedEvents:  # TODO: Technically entry can be unbound here.
                     pending.append(entry)
 
                 # Get list of events to discard
                 if not this.discardedEvents:
                     r = this.session.get('https://www.edsm.net/api-journal-v1/discard', timeout=_TIMEOUT)
                     r.raise_for_status()
                     this.discardedEvents = set(r.json())
-                    this.discardedEvents.discard('Docked')	# should_send() assumes that we send 'Docked' events
+                    this.discardedEvents.discard('Docked')  # should_send() assumes that we send 'Docked' events
                     assert this.discardedEvents			# wouldn't expect this to be empty
```

Comment:
Do we REALLY want an assert for this?  What if EDSM returns something that can cause this to be None?  Sounds more like a logger.error() and from this code condition to me.

Reply:
Annoyingly that is what this does. It bails out because of the try/accept. But agreed. Replacing with error and `continue`